include/extractor/raster_source.hpp

```diff
@@ -143,8 +144,32 @@ class RasterContainer
     RasterDatum GetRasterInterpolateFromSource(unsigned int source_id, double lon, double lat);
 
   private:
+};
+
+// << singletone >> RasterCache
+class RasterCache
+{
+  public:
+    // class method to get the instance
+    static RasterCache &getInstance()
+    {
+        if (NULL == g_instance)
+        {
+            g_instance = new RasterCache();
```

Comment:
Where this object will be deleted?

Reply:
Hi, thanks for your inquiry.
Singletone object will be deleted after main() is finished or exit() is called.
Ref: §3.6.3, in ISO/IEC 14882:2003

Reply:
Hi,
didn't know about such behavior.  Thank you for the reference.